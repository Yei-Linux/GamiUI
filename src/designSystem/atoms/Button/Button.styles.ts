import { css } from "@emotion/react";
import styled from "@emotion/styled";
import { ButtonType } from "../../../core/domain/types";
import { colorLight } from "../../../styles/theme";
import { twinStyles } from "../../../styles/utilities/twinStyles";

const ButtonTypes = (typeStyle: ButtonType) => css`
  background: ${colorLight.button[typeStyle].bg};
  color: ${colorLight.button[typeStyle].color};
  border: ${colorLight.button[typeStyle].border};
`;

export const ButtonWrapper = styled.button<{
  typeStyle?: ButtonType;
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  textAlign?: string;
  fontWeight?: string;
}>`
  border: none;
  outline: none;
  cursor: pointer;
  appearance: none;
  padding: 10px;

  ${(props: any) => twinStyles(props)};
  ${(props: any) => ButtonTypes(props.typeStyle)};
`;
