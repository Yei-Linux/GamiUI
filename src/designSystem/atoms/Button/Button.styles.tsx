import styled from '@emotion/styled'
import { mixinFlexVariants } from 'styles/mixins/flex'
import { setGenericPropStyles } from 'styles/utilities/genericPropStyles'
import {
  BorderType,
  ButtonType,
  FontWeightType,
  HeightType,
  ShadowType,
  TextAlignType,
  WidthType,
} from 'core/domain/types'
import { mixinComponentsTypes } from 'styles/mixins/componentsTypes'
import { spacing } from 'styles/tokens/spacing'

export const Button = styled.button<{
  $variant?: ButtonType
  $border?: BorderType
  $shadow?: ShadowType
  $width?: WidthType
  $heigth?: HeightType
  $textAlign?: TextAlignType
  $fontWeight?: FontWeightType
}>`
  ${mixinFlexVariants({ justifyContent: 'center', alignItems: 'center' })};

  border: none;
  outline: none;
  appearance: none;

  padding: ${spacing.padding.sm} ${spacing.padding.md};

  &:hover {
    cursor: pointer;
  }

  ${({ $variant }) =>
    mixinComponentsTypes('light', $variant || 'primary', 'button')};

  ${({ $border, $shadow, $fontWeight, $width, $textAlign, $heigth }) =>
    setGenericPropStyles({
      border: $border || 'sm',
      shadow: $shadow || 'md',
      width: $width || 'auto',
      heigth: $heigth || 'auto',
      fontWeight: $fontWeight,
      textAlign: $textAlign,
    })};
`
