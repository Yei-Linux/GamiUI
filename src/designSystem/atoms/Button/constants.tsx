export const options = {
  type: [
    "primary",
    "secondary",
    "tertiary",
    "cuaternary",
    "text",
    "link",
    "success",
    "warning",
    "danger",
    "info",
  ] as const,
};
