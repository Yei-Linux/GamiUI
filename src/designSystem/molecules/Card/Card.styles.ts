import styled from "@emotion/styled";
import { colorLight } from "../../../styles/theme";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const CardWrapper = styled.div<{
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
}>`
  width: auto;
  display: flex;
  flex-direction: column;
  overflow: hidden;
  min-width: 300px;
  min-height: 150px;

  ${(props: any) => twinStyles(props)};
`;

export const CardCover = styled.div<{}>`
  width: 100%;
  margin: 0px;
  border-radius: 0;
  display: flex;
  justify-content: center;
`;

export const CardContent = styled.div`
  padding: 1rem;
`;

export const CardFooter = styled.div`
  padding: 1rem;
  width: 100%;
  margin: 0px;
  border: 1px solid ${colorLight.neutral.seven};
`;
