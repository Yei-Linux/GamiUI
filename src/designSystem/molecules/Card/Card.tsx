import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { SubtitleContainer, TitleContainer } from "../../../styles/utilities/text";
import {
  CardContent,
  CardCover,
  CardFooter,
  CardWrapper,
} from "./Card.styles";

export interface ICard extends IGeneralProps {
  /**
   * Content Card to show
   */
  children: React.ReactNode;
}

export const Card = ({ children, ...args }: ICard) => (
  <CardWrapper {...args}>{children}</CardWrapper>
);

Card.Cover = ({ children }: { children: React.ReactNode }) => (
  <CardCover>{children}</CardCover>
);

Card.Content = ({
  title,
  description,
}: {
  title: React.ReactNode;
  description: React.ReactNode;
}) => (
  <CardContent>
    <TitleContainer>{title}</TitleContainer>
    <SubtitleContainer>{description}</SubtitleContainer>
  </CardContent>
);

Card.Footer = ({ children }: { children: React.ReactNode }) => (
  <CardFooter>{children}</CardFooter>
);

Card.defaultProps = {
  shadow: "SMALL",
  border: "MEDIUM",
  width: "NORMAL",
  height: "NORMAL",
};

export default Card;
