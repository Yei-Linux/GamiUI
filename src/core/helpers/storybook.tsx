import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};