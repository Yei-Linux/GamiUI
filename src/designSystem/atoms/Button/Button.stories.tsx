import { Meta } from "@storybook/react";
import {
  getListTemplate,
  stylesControl,
} from "../../../core/helpers/storybook";

import Button from ".";
import { options as optionsButton } from "./constants";
import { options as optionsStyles } from "../../../core/utils/constants";

const ListTemplate = getListTemplate(Button);

export default {
  title: "Atoms/Button",
  component: Button,
  argTypes: {
    children: { control: "text" },
    type: { control: { type: "select", options: optionsButton.type } },
    ...stylesControl,
  },
} as Meta;

export const TypesButton = ListTemplate.bind({});
TypesButton.args = {
  items: optionsButton.type.map((type) => ({ type })),
  field: 'type',
  children: "Text",
  width: "MEDIUM",
  heigth: "SMALL",
  shadow: "",
  border: "ROUNDED",
};

export const WidthButton = ListTemplate.bind({});
WidthButton.args = {
  items: optionsStyles.width.map((width) => ({ width })),
  field: 'width',
  children: "Text",
  heigth: "SMALL",
  shadow: "",
  border: "ROUNDED",
};
WidthButton.storyName = "Width Sizes";
