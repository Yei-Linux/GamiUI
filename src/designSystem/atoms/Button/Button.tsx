import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { ButtonType } from "../../../core/domain/types";
import { ButtonWrapper } from "./Button.styles";

export interface IButton extends IGeneralProps {
  /**
   * Content Button to show
   */
  children: React.ReactNode;
  /**
   * Button Type
   */
  type: ButtonType;
}

const Button = ({ children, type, ...args }: IButton) => {
  return (
    <ButtonWrapper {...args} typeStyle={type} type="submit">
      {children}
    </ButtonWrapper>
  );
};

Button.defaultProps = {
  type: "primary",
  border: "MEDIUM",
  heigth: "SMALL",
  width: "SMALL",
};

export default Button;
