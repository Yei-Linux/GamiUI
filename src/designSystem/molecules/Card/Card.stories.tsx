import React from "react";
import { Meta } from "@storybook/react";
import { stylesControl } from "../../../core/helpers/storybook";

import Card from ".";
import Button from "../../atoms/Button";

export default {
  title: "Molecules/Card",
  component: Card,
  args: { ...Card.defaultProps },
  argTypes: {
    ...stylesControl,
  },
} as Meta;

export const Basic = (args: any) => (
  <Card {...args}>
    <p style={{ padding: "1rem" }}>Hola como estas</p>
  </Card>
);

Basic.args = {};

export const Content = (args: any) => (
  <Card {...args}>
    <Card.Content
      title="Hi this is title"
      description="Hi this is description"
    />
  </Card>
);

Content.args = {};

export const Cover = (args: any) => (
  <Card {...args}>
    <Card.Cover>
      <img
        width="100%"
        src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
      />
    </Card.Cover>
    <Card.Content
      title="Hi this is title"
      description="Hi this is description"
    />
  </Card>
);

Cover.args = {};

export const Footer = (args: any) => (
  <Card {...args}>
    <Card.Cover>
      <img
        width="100%"
        src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
      />
    </Card.Cover>
    <Card.Content
      title="Hi this is title"
      description="Hi this is description"
    />
    <Card.Footer>
      <Button
        border="MEDIUM"
        heigth="SMALL"
        type="secondary"
        width="SMALL"
      >
        Text
      </Button>
    </Card.Footer>
  </Card>
);

Footer.args = {};
