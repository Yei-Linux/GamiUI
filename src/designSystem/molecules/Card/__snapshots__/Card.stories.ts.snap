// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Card Basic 1`] = `
<div
  className="css-1ohc8tn-CardWrapper e1ft0i6y0"
  height="NORMAL"
  width="NORMAL"
>
  <p
    style={
      Object {
        "padding": "1rem",
      }
    }
  >
    Hola como estas
  </p>
</div>
`;

exports[`Storyshots Molecules/Card Content 1`] = `
<div
  className="css-1ohc8tn-CardWrapper e1ft0i6y0"
  height="NORMAL"
  width="NORMAL"
>
  <div
    className="css-1keji-CardContent e1ft0i6y2"
  >
    <div
      className="css-144j0q-TitleContainer etn7vk90"
    >
      Hi this is title
    </div>
    <div
      className="css-92onx5-SubtitleContainer etn7vk91"
    >
      Hi this is description
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Card Cover 1`] = `
<div
  className="css-1ohc8tn-CardWrapper e1ft0i6y0"
  height="NORMAL"
  width="NORMAL"
>
  <div
    className="css-ebfyr-CardCover e1ft0i6y1"
  >
    <img
      src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
      width="100%"
    />
  </div>
  <div
    className="css-1keji-CardContent e1ft0i6y2"
  >
    <div
      className="css-144j0q-TitleContainer etn7vk90"
    >
      Hi this is title
    </div>
    <div
      className="css-92onx5-SubtitleContainer etn7vk91"
    >
      Hi this is description
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/Card Footer 1`] = `
<div
  className="css-1ohc8tn-CardWrapper e1ft0i6y0"
  height="NORMAL"
  width="NORMAL"
>
  <div
    className="css-ebfyr-CardCover e1ft0i6y1"
  >
    <img
      src="https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png"
      width="100%"
    />
  </div>
  <div
    className="css-1keji-CardContent e1ft0i6y2"
  >
    <div
      className="css-144j0q-TitleContainer etn7vk90"
    >
      Hi this is title
    </div>
    <div
      className="css-92onx5-SubtitleContainer etn7vk91"
    >
      Hi this is description
    </div>
  </div>
  <div
    className="css-ykbte7-CardFooter e1ft0i6y3"
  >
    <button
      className="css-1ru1eiv-ButtonWrapper ekfoxpm0"
      type="submit"
      width="SMALL"
    >
      Text
    </button>
  </div>
</div>
`;
